data-prepper-plugins/date-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/date/DateProcessor.java

```diff
@@ -71,7 +72,16 @@ public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
                 zonedDateTime =  getDateTimeFromTimeReceived(record);
 
             else if (keyToParse != null && !keyToParse.isEmpty()) {
-                zonedDateTime = getDateTimeFromMatch(record);
+                Pair<String, Instant> result = getDateTimeFromMatch(record);
+                if (result != null) {
+                    zonedDateTime = result.getLeft();
+                    Instant timeStamp = result.getRight();
+                    if (dateProcessorConfig.getToOriginationMetadata()) {
+                        Event event = (Event)record.getData();
+                        event.getMetadata().setExternalOriginationTime(timeStamp);
```

Comment:
I did not do that because I am not sure if we want to tie them together. What if we don't want external latency metric but set metadata only? This way, we could introduce another flag in date processor in future if such requirement arises.